sway-parse/src/module.rs

```diff
@@ -28,9 +29,30 @@ impl ParseToEnd for Module {
     fn parse_to_end<'a, 'e>(mut parser: Parser<'a, 'e>) -> ParseResult<(Self, ParserConsumed<'a>)> {
         let (kind, semicolon_token) = parser.parse()?;
         let mut dependencies = Vec::new();
-        while let Some(dep) = parser.guarded_parse::<DepToken, _>()? {
-            dependencies.push(dep);
+
+        // Parses multiple Dependency
+        loop {
+            // Return error if there is any DocComment before a Dependency
+            let mut doc_comment: Option<DocComment> = None;
+            let mut token_trees = parser.token_trees();
+            while let Some((doc, tokens)) = Peeker::with::<DocComment>(token_trees) {
+                token_trees = tokens;
+                doc_comment = Some(doc);
+            }
+            if let Some(doc) = doc_comment {
+                if let Some((_, _)) = Peeker::with::<DepToken>(token_trees) {
+                    return Err(parser
+                        .emit_error_with_span(ParseErrorKind::CannotDocCommentDepToken, doc.span));
+                }
+            }
+
+            if let Some(dep) = parser.guarded_parse::<DepToken, _>()? {
+                dependencies.push(dep);
+            } else {
+                break;
+            }
         }
+
```

Comment:
Thank you for reviewing this PR. While implementing the changes you mentioned I noticed that dependencies modules are parsed before `convert_parse_tree` is called. [sway-core/src/lib.rs#L157](https://github.com/FuelLabs/sway/blob/11496e6cedd0cacc30c461440b22d854eb3def58/sway-core/src/lib.rs#L157) So it requires the dependency information before `convert_parse_tree`, not sure if there is a workaround this.

Reply:
> So here you're introducing infinite look-ahead but also not adding any error recovery in case there was a doc comment.

If there is doc comment it should be consumed by `let (items, consumed) = parser.parse_to_end()?;`

@Centril do you want me to error handling when there is a doc comment but no items?

I would do the changes you mentioned but I am currently blocked by the fact that dependencies are required before `convert_parse_tree` is called. So there is no clean way of getting them at [sway-core/src/lib.rs#L157](https://github.com/FuelLabs/sway/blob/11496e6cedd0cacc30c461440b22d854eb3def58/sway-core/src/lib.rs#L157)